### PR TITLE
Add "Raidraptor Readiness (Anime)"

### PR DESCRIPTION
**Pull request by idk-what-to-name-here**
https://yugipedia.com/wiki/Raidraptor_-_Readiness_(anime)
- [X] I am following the [contributing guidelines](https://github.com/ProjectIgnis/CardScripts/blob/master/CONTRIBUTING.md).
- [X] I am making the [changes to modernize the scripts](https://github.com/ProjectIgnis/CardScripts/blob/master/MODERNIZING.md).

**New card checklist**

- [X] I have submitted a corresponding database entry for [BabelCDB](https://github.com/ProjectIgnis/BabelCDB/blob/master/README.md) according to its guidelines. Link to the pull request: _https://github.com/larry126/BabelCDB/pull/122_
- [X] This card is not marked with a reason to not be added in the unofficial card tracker in `#card-scripting-101`.
Supervising staff member(s): _larry126_

### CODE REVIEW

**Comment by larry126** [comment]
- Could just use `aux.damcon1` for e2 and e3
- For e1, check out how Kuriboh is scripted
- In s.damop, instead of using `e:GetHandler()` repeatedly, create a variable `c` that holds the value of `e:GetHandler()` and use `c` instead
- In location, reason, and reset, use `|` instead of `+`

File: unofficial/c511027787.lua [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	local e2=Effect.CreateEffect(e:GetHandler())
+	e2:SetType(EFFECT_TYPE_FIELD)
+	e2:SetCode(EFFECT_AVOID_BATTLE_DAMAGE)
+	e2:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+	e2:SetTargetRange(1,0)
+	e2:SetValue(1)
+	e2:SetReset(RESET_PHASE+PHASE_END)
+	Duel.RegisterEffect(e2,tp)
+	local e3=Effect.CreateEffect(e:GetHandler())
+	e3:SetProperty(EFFECT_FLAG_PLAYER_TARGET+EFFECT_FLAG_CLIENT_HINT+EFFECT_FLAG_OATH)
```
Comment (larry126) [inline unofficial/c511027787.lua:77]: Not sure which card you copied this effect from, but it is not complete and wrong.
Not complete: This effect was used to add a client hint to the activator to notify both players that the activator is not taking Battle Damage for the rest of the turn, you can tell this from `EFFECT_FLAG_CLIENT_HINT` flag. And it's currently missing the SetDescription that contains the actual client hint message.
Wrong: It should not have EFFECT_FLAG_OATH here, this flag is often used in cost for things like `If you **activate** this effect, you cannot do X for the rest of the turn`. For a restriction like this, if you negate the effect's **activation**, the restriction will no longer apply to you, and the flag `EFFECT_FLAG_OATH` is used to handle this interaction automatically. And clearly this is not relevant to Readiness.

Additionally, to add the player hint message, you could just add the flag `EFFECT_FLAG_CLIENT_HINT` to the e2 above, and add the corresponding SetDescription to it.

File: unofficial/c511027787.lua [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+--Raidraptor - Readiness (Anime)
+--Made by When
+local s,id=GetID()
+function s.initial_effect(c)
+    --Activate (if you would take battle damage)
+	local e1=Effect.CreateEffect(c)
+	e1:SetDescription(aux.Stringid(id,0))
+	e1:SetType(EFFECT_TYPE_ACTIVATE)
+	e1:SetCode(EVENT_PRE_DAMAGE_CALCULATE)
+	e1:SetCondition(s.condition1)
```
Comment (larry126) [inline unofficial/c511027787.lua:11]: Could just use `aux.damcon1` here

File: unofficial/c511027787.lua [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	e1:SetCode(EFFECT_CHANGE_DAMAGE)
+	e1:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+	e1:SetTargetRange(1,0)
+	e1:SetLabel(cid)
+	e1:SetValue(s.refval)
+	e1:SetReset(RESET_CHAIN)
+	Duel.RegisterEffect(e1,tp)
+end
+function s.damcon(e,tp,eg,ep,ev,re,r,rp)
+	return aux.damcon1(e,tp,eg,ep,ev,re,r,rp)
```
Comment (larry126) [inline unofficial/c511027787.lua:51]: You need to have a LP that is not 10 to activate this effect

File: unofficial/c511027787.lua [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	e1:SetCondition(s.condition1)
+	e1:SetOperation(s.activate1)
+	c:RegisterEffect(e1)
+	--Activate (if you would take effect damage)
+	local e2=Effect.CreateEffect(c)
+	e2:SetDescription(aux.Stringid(id,0))
+	e2:SetProperty(EFFECT_FLAG_DAMAGE_STEP)
+	e2:SetType(EFFECT_TYPE_ACTIVATE)
+	e2:SetCode(EVENT_CHAINING)
+	e2:SetCondition(s.condition2)
```
Comment (larry126) [inline unofficial/c511027787.lua:20]: Could just use `aux.damcon1` here

File: unofficial/c511027787.lua [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	e2:SetOperation(s.activate2)
+	c:RegisterEffect(e2)
+	--Banish From GY
+	local e3=Effect.CreateEffect(c)
+	e3:SetDescription(aux.Stringid(id,1))
+	e3:SetProperty(EFFECT_FLAG_DAMAGE_STEP)
```
Comment (larry126) [inline unofficial/c511027787.lua:26]: This effect cannot be used during the damage step, so no EFFECT_FLAG_DAMAGE_STEP

**Review by larry126 — CHANGES_REQUESTED** [review]

**Comment by idk-what-to-name-here** [comment]
done

**Pull request merged**